### PR TITLE
add `databricks_artifact_allowlist`

### LINKED ISSUE #2688: [Unity Catalog Allow List] Add integration for UC Artifact Allow List
There is a new feature in Unity Catalog to allow list jars and init scripts. It would be great to include support for this in the provider. The new API is located here https://docs.databricks.com/api/workspace/artifactallowlists/update

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- Add new resource `databricks_artifact_allowlist` to support [Unity Catalog allowlist](https://docs.databricks.com/en/data-governance/unity-catalog/manage-privileges/allowlist.html)
- Add `common.NoCustomize` function to simplify schema creation
- 
close #2688

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Review by alexott — APPROVED** [review]
looks good for me

**Comment by alexott** [comment]
@mgyucht we got multiple requests about it already...

File: docs/resources/artifact_allowlist.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+subcategory: "Unity Catalog"
+---
+# databricks_artifact_allowlist Resource
```
[the remark this answers is not in the document]
Comment (mgyucht) [reply docs/resources/artifact_allowlist.md:4]: The rule of thumb is that only metastore and metastore assignment are account-level operations, and that everything else is workspace-level.

[the remark this answers is not in the document]
Comment (nkvuong) [reply docs/resources/artifact_allowlist.md:4]: added a note

File: catalog/resource_artifact_allowlist_test.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+}
+
+func TestArtifactAllowlistUpdate(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:          http.MethodPut,
+				Resource:        "/api/2.1/unity-catalog/artifact-allowlists/INIT_SCRIPT",
+				ExpectedRequest: updateArtifact,
+				Response:        artifactInfo,
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply catalog/resource_artifact_allowlist_test.go:181]: fixed test

**Review by mgyucht — APPROVED** [review]
I'm going to revisit the schema definitions soon to make it easier to define schemas based on different SDK operations. In the meantime, let's go ahead with this change.

**Pull request merged**